webapi/src/main/scala/org/knora/webapi/responders/v2/StandoffResponderV2.scala

```diff
@@ -0,0 +1,781 @@
+
+                _ = if (existingMappingResponse.statements.nonEmpty) {
+                    throw BadRequestException(s"mapping IRI $mappingIri already exists")
+                }
+
+                createNewMappingSparql = queries.sparql.v2.txt.createNewMapping(
+                    triplestore = settings.triplestoreType,
+                    dataNamedGraph = namedGraph,
+                    mappingIri = mappingIri,
+                    label = label,
+                    defaultXSLTransformation = defaultXSLTransformation,
+                    mappingElements = mappingElements
+                ).toString()
+
+                // Do the update.
+                createResourceResponse: SparqlUpdateResponse <- (storeManager ? SparqlUpdateRequest(createNewMappingSparql)).mapTo[SparqlUpdateResponse]
+
+                // check if the mapping has been created
+                newMappingResponse <- (storeManager ? SparqlConstructRequest(getExistingMappingSparql)).mapTo[SparqlConstructResponse]
+
+                _ = if (newMappingResponse.statements.isEmpty) {
+                    log.error(s"Attempted a SPARQL update to create a new resource, but it inserted no rows:\n\n$newMappingResponse")
+                    throw UpdateNotPerformedException(s"Resource $mappingIri was not created. Please report this as a possible bug.")
+                }
+
+                // get the mapping from the triplestore and cache it thereby
+                _ = getMappingFromTriplestore(mappingIri, userProfile)
+
+
+            } yield {
+                CreateMappingResponseV2(
+                    mappingIri = mappingIri,
+                    label = label,
+                    projectIri = projectIri
+                )
+            }
+
+
+            createMappingFuture.recover {
+                case validationException: SAXException => throw BadRequestException(s"the provided mapping is invalid: ${validationException.getMessage}")
+
+                case ioException: IOException => throw NotFoundException(s"The schema could not be found")
+
+                case unknown: Exception => throw BadRequestException(s"the provided mapping could not be handled correctly: ${unknown.getMessage}")
+
+            }
+
+        }
+
+        for {
+            // Don't allow anonymous users to create a mapping.
+            // Don't allow anonymous users to create a mapping.
+            userIri: IRI <- Future {
+                if (userProfile.isAnonymousUser) {
+                    throw ForbiddenException("Anonymous users aren't allowed to create mappings")
+                } else {
+                    userProfile.id
+                }
+            }
+
+            // check if the given project IRI represents an actual project
+            projectInfo: ProjectInfoResponseV1 <- (responderManager ? ProjectInfoByIRIGetRequestV1(
```

Comment:
Please make this use messages from `ProjectsMessagesADM.scala` instead of v1 messages.

Reply:
done in https://github.com/dhlab-basel/Knora/pull/811/commits/2d8e33d09b0c77f5916ef1fb3e6aabe1e9fb61ea